Library/ViewModels/DiscoveryFiltersViewModel.swift

```diff
@@ -316,10 +304,47 @@ private func favorites(selectedRow: SelectableRow, categories: [KsApi.Category])
   return faves.isEmpty ? nil : faves
 }
 
-private func cachedCategories() -> [KsApi.Category]? {
-  return AppEnvironment.current.cache[KSCache.ksr_discoveryFiltersCategories] as? [KsApi.Category]
+private func cachedCategories() -> [RootCategoriesEnvelope.Category]? {
+  return AppEnvironment.current
+    .cache[KSCache.ksr_discoveryFiltersCategories] as? [RootCategoriesEnvelope.Category]
 }
 
-private func cache(categories: [KsApi.Category]) {
+private func cache(categories: [RootCategoriesEnvelope.Category]) {
   AppEnvironment.current.cache[KSCache.ksr_discoveryFiltersCategories] = categories
 }
+
+private let rootCategoriesQuery: NonEmptySet<Query> = Query.rootCategories(
+  .id +| [
+    .name,
+    .subcategories(
+      [],
+      .totalCount +| [
+        .nodes(
+          .id +| [
+            .name
+          ]
+        )
+      ]
+    )
+  ]
+  ) +| []
+
+private struct RootCategoriesEnvelope: Swift.Decodable {
+  let rootCategories: [Category]
+
+   struct Category: Swift.Decodable {
+    let id: String
+    let name: String
+    let subcategories: SubcategoryConnection
+
+     struct SubcategoryConnection: Swift.Decodable {
+      let totalCount: Int
+      let nodes: [Node]
```

Comment:
Might be able to generalize `Node` to `Node<T>` for better readability when using nodes in other places like `Node<Category>` or `Node<Project>`. I believe it'll still work with `Swift.Decodable`?

Reply:
It makes sense. Yes, it will work with Swift.Decodable if we declare node as Node<T: Swift.Decodable>